src/dos/drive_local.cpp

```diff
@@ -42,28 +42,34 @@ bool localDrive::FileCreate(DOS_File * * file,char * name,Bit16u /*attributes*/)
 	bool existing_file = false;
 	
 	FILE * test = fopen_wrap(temp_name,"rb+");
-	if(test) {
+	if (test) {
 		fclose(test);
 		existing_file=true;
 
 	}
 	
 	FILE * hand = fopen_wrap(temp_name,"wb+");
-	if (!hand){
+	if (!hand) {
 		LOG_MSG("Warning: file creation failed: %s",newname);
 		return false;
 	}
    
-	if(!existing_file) dirCache.AddEntry(newname, true);
+	if (!existing_file) dirCache.AddEntry(newname, true);
 	/* Make the 16 bit device information */
 	*file=new localFile(name,hand);
 	(*file)->flags=OPEN_READWRITE;
 
 	return true;
 }
 
-bool localDrive::FileOpen(DOS_File * * file,char * name,Bit32u flags) {
-	const char* type;
+bool localDrive::isNewWriteProtectedFile(const std::string& filename) {
+	const std::pair<std::set<std::string>::iterator, bool> \
+		ret(write_protected_files.insert(filename));
```

Comment:
The filename here is the absolute path in the filesystem, which isn't ideal for comparing in a set /because/the/prefix/test/is/always/the/same/<unique-filename.dat>

So I would have liked to have two functions: GetCWD() and MakeRelativePath(cwd, filename), so I could simply store the relative path to these files.

But alas, getting CWD apparently isn't portable.. so it would have been a mess of code and ifdef's for each OS.   What a mess.  Fortunately C++17 comes to the rescue with a bunch of filesystem metadata calls like this; so eventually we'll have this!